From b72d905836bc083ebf2af95cf8d47274c3d2d56b Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Tue, 10 Dec 2024 16:36:25 +0800
Subject: [PATCH 237/301] AOSCOS: MIPS: audit: declare function prototypes for
 audit_classify_syscall_*()

Functions `audit_classify_syscall_n32()', `audit_classify_syscall_o32()`
were never declared in the previous code and causes build errors due to
`-Werror=missing-prototype'.

Create function prototype declarations in `audit-{n,o}32.h' and reference
them in each of the respective sources. Also remove the extern references
as they are no longer needed.

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 arch/mips/kernel/audit-n32.c    | 1 +
 arch/mips/kernel/audit-n32.h    | 5 +++++
 arch/mips/kernel/audit-native.c | 5 ++---
 arch/mips/kernel/audit-o32.c    | 1 +
 arch/mips/kernel/audit-o32.h    | 5 +++++
 5 files changed, 14 insertions(+), 3 deletions(-)
 create mode 100644 arch/mips/kernel/audit-n32.h
 create mode 100644 arch/mips/kernel/audit-o32.h

diff --git a/arch/mips/kernel/audit-n32.c b/arch/mips/kernel/audit-n32.c
index 2248badc3acb..2e9ac29588c8 100644
--- a/arch/mips/kernel/audit-n32.c
+++ b/arch/mips/kernel/audit-n32.c
@@ -4,6 +4,7 @@
 #include <linux/types.h>
 #include <linux/audit.h>
 #include <asm/unistd.h>
+#include "audit-n32.h"
 
 static unsigned dir_class_n32[] = {
 #include <asm-generic/audit_dir_write.h>
diff --git a/arch/mips/kernel/audit-n32.h b/arch/mips/kernel/audit-n32.h
new file mode 100644
index 000000000000..781de2e9ad66
--- /dev/null
+++ b/arch/mips/kernel/audit-n32.h
@@ -0,0 +1,5 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
+
+#ifdef CONFIG_AUDITSYSCALL_N32
+int audit_classify_syscall_n32(int abi, unsigned syscall);
+#endif /* CONFIG_AUDITSYSCALL_N32 */
diff --git a/arch/mips/kernel/audit-native.c b/arch/mips/kernel/audit-native.c
index 09ae3dbcfecb..88e283f8df38 100644
--- a/arch/mips/kernel/audit-native.c
+++ b/arch/mips/kernel/audit-native.c
@@ -2,6 +2,8 @@
 #include <linux/types.h>
 #include <linux/audit.h>
 #include <asm/unistd.h>
+#include "audit-n32.h"
+#include "audit-o32.h"
 
 static unsigned dir_class[] = {
 #include <asm-generic/audit_dir_write.h>
@@ -37,9 +39,6 @@ int audit_classify_arch(int arch)
 	return 0;
 }
 
-extern int audit_classify_syscall_o32(int abi, unsigned syscall);
-extern int audit_classify_syscall_n32(int abi, unsigned syscall);
-
 int audit_classify_syscall(int abi, unsigned syscall)
 {
 	int res;
diff --git a/arch/mips/kernel/audit-o32.c b/arch/mips/kernel/audit-o32.c
index e8b9b50f7d26..6bf302c04e5d 100644
--- a/arch/mips/kernel/audit-o32.c
+++ b/arch/mips/kernel/audit-o32.c
@@ -4,6 +4,7 @@
 #include <linux/types.h>
 #include <linux/audit.h>
 #include <linux/unistd.h>
+#include "audit-o32.h"
 
 static unsigned dir_class_o32[] = {
 #include <asm-generic/audit_dir_write.h>
diff --git a/arch/mips/kernel/audit-o32.h b/arch/mips/kernel/audit-o32.h
new file mode 100644
index 000000000000..8a782fdd1c6e
--- /dev/null
+++ b/arch/mips/kernel/audit-o32.h
@@ -0,0 +1,5 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
+
+#ifdef CONFIG_AUDITSYSCALL_O32
+int audit_classify_syscall_o32(int abi, unsigned syscall);
+#endif /* CONFIG_AUDITSYSCALL_O32 */
-- 
2.48.1

